infrastructures/docker/docker-client/src/main/java/org/eclipse/che/infrastructure/docker/client/DockerConnector.java

```diff
@@ -900,7 +907,7 @@ private String buildImage(
                       "Docker image build failed. Image id not found in build output.", 500);
                 });
 
-        return imageIdFuture.get();
+        return imageIdFuture.get(imageBuildTimeoutSec, TimeUnit.SECONDS);
```

Comment:
Do we interrupt Thread which builds images? Will be building really interrupted?

Reply:
:O good point, @akorneta are you sure that docker build process will be interrupted?

Reply:
@sleshchenko Do you know a better approach? 
AFAIK from recent versions of Docker, it is a way to perform interruption of the build.

Reply:
@garagatyi No, I don't. I just ask if it will solve the original issue :smiley_cat:

Reply:
AFAIK Docker CLI does the same

Reply:
Ok, thanks for explanation

Reply:
@sleshchenko , @riuvshin so if you carefully look at the implementation of docker image build you will see that in the case when exceptions occur in main(waiting) thread there no any additional work for interruption of building thread the explanation why is placed [here](https://github.com/eclipse/che/blob/19465ec962c4ec4676a1c7aa7d39393c3e24f4a2/infrastructures/docker/docker-client/src/main/java/org/eclipse/che/infrastructure/docker/client/DockerConnector.java#L881-L887).